src/components/DownloadContactListButton.tsx

```diff
@@ -1,25 +1,34 @@
+
 import React from 'react'
 import { UnprocessedResultsFromCRM } from '../types'
-
 type DownloadButtonProps = {
     results: UnprocessedResultsFromCRM[]
 }
+
 export function DownloadContactListButton (props: DownloadButtonProps) {
     let downloadUrl = null
     let csvData = '"Property Address","Property Type (Marketing)","Owner","Owner Mobile","Owner Phone","Contact","Contact Mobile","Contact Work Phone"\r\n'
 
-    props.results.forEach((result) => {
+    const objectArr = props.results
+    function getUniqueListBy (arr: UnprocessedResultsFromCRM[], key: string) {
+        return [...new Map(arr.map((propertyObject: any) => [propertyObject[key], propertyObject])).values()]
+    }
```

Comment:
Put the function definition outside of the loop otherwise it'll be [defining it more than once](https://stackoverflow.com/questions/20896130/javascript-best-practices-wheres-the-best-place-to-define-a-helper-function-i).

Reply:
Isn't it outside of the loop?

Reply:
Opps, sorry. I missed the change of the forEach to another part of the code.